dev/src/transaction.ts

```diff
@@ -419,10 +415,10 @@ export class Transaction {
  * @param documentRefsOrReadOptions An array of document references followed by
  * an optional ReadOptions object.
  */
-export function parseGetAllArguments(
-  documentRefsOrReadOptions: Array<DocumentReference | ReadOptions>
-): {documents: DocumentReference[]; fieldMask: FieldPath[] | null} {
-  let documents: DocumentReference[];
+export function parseGetAllArguments<T>(
+  documentRefsOrReadOptions: Array<DocumentReference<T> | ReadOptions>
+): {documents: Array<DocumentReference<T>>; fieldMask: FieldPath[] | null} {
+  let documents: Array<DocumentReference<T>>;
```

Comment:
The way `getAll()` is implemented, it should be straightforward to support references of different types. Can you see if we can add an overload that supports `any`? We may be able to just redeclare `getAll<T>` as `getAll<T = any>`, as long as the compiler still types `getAll(doc<Foo>, doc<Foo>)` as returning `DocSnap<Foo>`.

Reply:
After our conversation, we've decided not to cast to `any` by default since the user would have to cast to `<any>` themselves anyways.